### DISCUSSION TITLE
"The Salt Master has cached the public key for this node" after renaming minion/ip change

### DISCUSSION THREAD

**Discussion by linuxmoose** [uyuni-project/uyuni#6943] [Devel]
I have a CentOS 7.x system that was renamed.  Before renaming, I did "systemctl stop venv-salt-minion;systemctl disable venv-salt-minion", deleted it from the Uyuni web UI, edited /etc/salt/minion_id to reflect the new name, renamed/changed ip of the system, and restarted.  After I verified all was good with the system, I restarted venv-salt-minon.  The status shows "The Salt Master has cached the public key for this node, this salt minion will wait for 10 seconds before attempting to re-authenticate".  
I stopped venv-salt-minion, and tried the solution that has worked in the past for me:
rm -f /etc/machine-id
rm -f /var/lib/dbus/machine-id
dbus-uuidgen --ensure
systemd-machine-id-setup
Unfortunately, doing this and restarting venv-salt-minion yielded the same result.  I get the same error about the cached public key, and it does not show up in the UI.
Any suggestions?
Thanks.
Allen B.

**Comment by digdilem-work** [comment]
Hi Allen,

Are there any blocked/red keys showing in the webui -> Salt -> Keys ?  If so and it matches your machine, remove them from there and try to rejoin.

(You’ve already renewed the machine-id, which was also a suggestion)

I’d then try removing all traces of venv-salt-minion (and salt-minion if that exists from an earlier install) and also delete /etc/venv-salt-minion/ and /etc/salt-minion  before retrying a fresh bootstrap.

By the way, for future reference, you can rename machines in Uyuni using the “System Properties (Edit These Properties)” link. The minion-id will remain on the previous hostname, but I think that’s purely cosmetic. A hardware refresh will update the changed IP (We do this often and it’s not caused us any issues)

Good luck,

Simon

From: Allen Beddingfield ***@***.***>
Sent: Wednesday, April 26, 2023 8:13 PM
To: uyuni-project/uyuni ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [uyuni-project/uyuni] "The Salt Master has cached the public key for this node" after renaming minion/ip change (Discussion #6943)


I have a CentOS 7.x system that was renamed. Before renaming, I did "systemctl stop venv-salt-minion;systemctl disable venv-salt-minion", deleted it from the Uyuni web UI, edited /etc/salt/minion_id to reflect the new name, renamed/changed ip of the system, and restarted. After I verified all was good with the system, I restarted venv-salt-minon. The status shows "The Salt Master has cached the public key for this node, this salt minion will wait for 10 seconds before attempting to re-authenticate".
I stopped venv-salt-minion, and tried the solution that has worked in the past for me:
rm -f /etc/machine-id
rm -f /var/lib/dbus/machine-id
dbus-uuidgen --ensure
systemd-machine-id-setup
Unfortunately, doing this and restarting venv-salt-minion yielded the same result. I get the same error about the cached public key, and it does not show up in the UI.
Any suggestions?
Thanks.
Allen B.
